src/openrct2/ride/Ride.cpp

```diff
@@ -538,7 +538,7 @@ bool TrackBlockGetNextFromZero(
 
         const auto& nextTrackCoordinate = ted.coordinates;
         uint8_t nextRotation = tileElement->GetDirectionWithOffset(nextTrackCoordinate.rotationBegin)
-            | (nextTrackCoordinate.rotationBegin & TRACK_BLOCK_2);
+            | (TrackPieceDirectionIsDiagonal(nextTrackCoordinate.rotationBegin) * kDirectionDiagonalMask);
```

Comment:
Oh shit, I misread. Hold on while I get a grip :')

Reply:
Could you just replace `& TRACK_BLOCK_2` with `& kDirectionDiagonalMask`? I misread the initial changes, sorry about that.